torch/distributed/_tensor/__init__.py

```diff
@@ -20,6 +20,97 @@
 ]
 
 
+def _dtensor_init_helper(init_op, *size, **kwargs) -> DTensor:
+    # get default device mesh if there's nothing specified
+    device_mesh = kwargs.pop("device_mesh", None)
+    if device_mesh is None:
+        device_mesh = get_global_device_mesh()
+
+    # set default placements to replicated if not specified
+    placements = kwargs.pop("placements", None)
+    if placements is None:
+        placements = [Replicate() for _ in range(device_mesh.ndim)]
+
+    # check device_mesh againts placements
+    assert device_mesh.ndim == len(
+        placements
+    ), "mesh dimension does not match the length of placements"
+
+    # normalize the size argument
+    if len(size) == 1 and isinstance(size[0], Sequence):
+        torch_size = size[0]
+    else:
+        torch_size = list(size)
+    torch_size = torch.Size(torch_size)
+    assert kwargs["layout"] == torch.strided, "layout value not supported!"
+    torch_stride = torch._prims_common.make_contiguous_strides_for(torch_size)
+
+    # get local tensor shape
+    local_shape = compute_local_shape(torch_size, device_mesh, placements)
+    # initialize the local tensor
+    if len(local_shape) == 0:
+        local_tensor = torch.empty(0, **kwargs)
+    else:
+        local_tensor = init_op(local_shape, **kwargs)
+
+    return DTensor(
+        local_tensor=local_tensor,
+        device_mesh=device_mesh,
+        placements=placements,
+        shape=torch_size,
+        dtype=local_tensor.dtype,
+        stride=torch_stride,
+        requires_grad=kwargs["requires_grad"],
+    )
+
+
+def ones(
+    *size,
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
```

Comment:
I am not quite sure if we want to expose the `device` argument and how it should behave, maybe we should remove it first from both `zeros` and `ones` to avoid confusion to the user.

Reply:
so basically we simply use the `device` from `device_mesh`?

Reply:
Yes, because exposing `device` to user is quite confusing, when creating DeviceMesh and setting up distributed environment (i.e. NCCL pg), user should already select the GPU device on each rank (if not nccl wouldn't work). If user specify a specific CUDA device that is different from the selected one, that might be confusing to us too, so we should probably remove this arg.